### PR TITLE
Moving over to the monitoring library

### PR DESCRIPTION
**Pull request by andrewrlee**
There's a new set of [shared libraries](https://github.com/ministryofjustice/hmpps-typescript-lib) that should allow us to share code more efficiently than current mechanisms.

The first of these is a [new library](https://github.com/ministryofjustice/hmpps-typescript-lib/tree/main/packages/monitoring) that wraps the 3 common endpoints used for monitoring and strives to ensure that teams have set up health monitoring of their dependencies.

The library will attempt to self install itself if you run it via npx: `npx @ministryofjustice/hmpps-monitoring`

It will then prompt you to perform some manual tasks - if you have stub tests for your health endpoints you might need add some additional stubbing.

### CODE REVIEW

File: server/middleware/setUpHealthChecks.ts [hunk 1]
```diff
@@ -1,39 +1,22 @@
...
 
-  router.get('/health', (req, res, next) => {
-    healthcheck(applicationInfo, result => {
-      if (result.status !== 'UP') {
-        res.status(503)
-      }
-      res.json(result)
-    })
-  })
+export default function setUpHealthChecks(applicationInfo: ApplicationInfo, components: ApiToMonitor[]): Router {
```
Comment (Thomas-Geraghty) [inline server/middleware/setUpHealthChecks.ts:9]: Where does type `ApiToMonitor` come from? Is it missing from this PR? 😃

Comment (andrewrlee) [reply server/middleware/setUpHealthChecks.ts:9]: it's on line 7^

Comment (andrewrlee) [reply server/middleware/setUpHealthChecks.ts]: Tidied this up to remove the extra type

File: .github/workflows/pipeline.yml [hunk 2]
```diff
@@ -27,18 +27,26 @@ jobs:
     name: node build
     uses: ministryofjustice/hmpps-github-actions/.github/workflows/node_build.yml@v1 # WORKFLOW_VERSION
     secrets: inherit
+    with:
+      node_version: 22.11
```
Comment (ushkarev) [inline .github/workflows/pipeline.yml:31]: i'd suggest simply `22` to not have to keep changing this number

Comment (andrewrlee) [reply .github/workflows/pipeline.yml:31]: yes, makes sense.. 
I'll probably remove this file and include the fix in a separate change

File: package.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "hmpps-template-typescript",
   "version": "0.0.1",
-  "description": "HMPPS Typescript Template",
+  "description": "HMPPS Typescriptmplate",
```
[the remark this answers is not in the document]
Comment (andrewrlee) [reply package.json:4]: using this ticket to test issues with github actions

**Review by psoleckimoj — APPROVED** [review]
LGTM!

**Review by ushkarev — APPROVED** [review]

**Review by ushkarev — APPROVED** [review]

**Pull request merged**